Add test when there are no segments

lucylq opened the pull request:
Summary:
The deprecation warning should only appear when there are values inside the constant_buffer and nothing inside the constant_segment.

Even when there are no constants in the program, we expect to have one 'segment' storing the placeholder for non-constants.

Fix failing jobs caused by error logging from #12295
Test failures: https://hud.pytorch.org/pr/pytorch/executorch/12631#46239138027

Arm test is using a recently-generated PTE, however hitting the deprecation warning because the pte has no constants.

Differential Revision: D78568839

digantdesai commented on runtime/executor/program.cpp:
```diff
@@ -217,18 +217,21 @@ Result<executorch_flatbuffer::ExecutionPlan*> get_execution_plan(
+      // https://github.com/pytorch/executorch/issues.
+      ET_LOG(
+          Error,
+          "!!DEPRECATED!! This branch is deprecated from ExecuTorch 0.7; re-export this PTE file to ensure support on newer runtimes.");
```
Nit: can we explain this a bit more and perhaps how to mitigate it? Seems like one has to read code to get around this error.

lucylq replied:
I previously had a longer message, but this caused errors in internal tests where they have a limit of 128 bytes for the log string - see [D77965467](https://www.internalfb.com/diff/D77965467?entry_point=16)

lucylq committed:
Do not load segment when only placeholder exists (#12646)

Summary:

The deprecation warning should only appear when there are values inside the constant_buffer and nothing inside the constant_segment.

Even when there are no constants in the program, we expect to have one 'DataSegment' storing the placeholder for non-constants. Move this into the 'if' branch.

Fix failing jobs caused by error logging from #12295
Test failures: https://hud.pytorch.org/pr/pytorch/executorch/12631#46239138027

Arm test is using a recently-generated PTE, however hitting the deprecation warning because the pte has no constants.

Reviewed By: larryliu0820

Differential Revision: D78568839

lucylq commented:
@pytorchbot cherry-pick --onto release/0.7 -c {release; add log for deprecated feature}

lucylq commented:
@pytorchbot cherry-pick --onto release/0.7 -c {release}

lucylq commented:
@pytorchbot cherry-pick --onto release/0.7 -c release
